corehq/apps/userreports/reports/sum_when_templates.py

```diff
@@ -0,0 +1,104 @@
+import re
+
+from dimagi.ext.jsonobject import (
+    IntegerProperty,
+    JsonObject,
+    ListProperty,
+    StringProperty,
+)
+
+from corehq.apps.userreports.specs import TypeProperty
+
+
+class SumWhenTemplateSpec(JsonObject):
+    type = StringProperty(required=True)
+    expression = StringProperty(required=True)
+    binds = ListProperty()
+    then = IntegerProperty()
+
+    def bind_count(self):
+        return len(re.sub(r'[^?]', '', self.expression))
+
+
+### Templates for testing
+
+class YearRangeTemplateSpec(SumWhenTemplateSpec):
+    type = TypeProperty('year_range')
+    expression = "year >= ? and year < ?"
+
+
+class UnderXMonthsTemplateSpec(SumWhenTemplateSpec):
+    type = TypeProperty("under_x_months")
+    expression = "age_at_registration < ?"
+
+
+### Templates for ICDS
```

Comment:
hmm. unfortunately it looks like we have a way for every other type of custom expression( https://github.com/dimagi/commcare-hq/blob/master/settings.py#L1937), but not these